src/codeu/chat/client/ClientMessage.java

```diff
@@ -202,6 +210,16 @@ private Uuid getCurrentTailMessageId() {
     }
     return nextMessageId;
   }
+  
+  // Map keywords in each message to each message that contains it
+  // Currently repeats each time updateMessages is called
+  public void mapKeywords(Message addMessage){
+	  String[] wordsArray = addMessage.content.split(" ");
```

Comment:
What about punctuations?

Take this message:
"This is really good, isn't it?"
This will be split up as:
This
is
really
good,
isn't it?

You need to split by punctuations as well.  I think you should use a regular expression to split the string. It will make it more robust.